api/v1alpha3/azurecluster_types.go

```diff
@@ -64,7 +64,7 @@ type AzureClusterStatus struct {
 	// A FailureDomain maps to Availability Zone with an Azure Region (if the region support them). An
 	// Availability Zone is a separate data center within a region and they can be used to ensure
 	// the cluster is more resilient to failure.
-	// See: https://docs.microsoft.com/en-us/azure/availability-zones/az-overview
+	// See: https://learn.microsoft.com/en-us/azure/reliability/availability-zones-overview
```

Comment:
When modifying these URLs, consider removing the `/en-us` part. It's a more readable URL and will redirect to a version localized to your browser's language (if available).

https://learn.microsoft.com/azure/reliability/availability-zones-overview

Reply:
@mboersma thanks, I appended the fix to the old doc lint fixing [commit](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2773/commits/353a39413d8be27d0bba1cb8aa4a383fc806dc13) and should be fine now